src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceDriveIfAvailable.scala

```diff
@@ -27,13 +27,9 @@ class ModeChoiceDriveIfAvailable(val beamServices: BeamServices) extends ModeCho
     val containsDriveAlt = alternatives.zipWithIndex.collect {
       case (trip, idx) if trip.tripClassifier == CAR => idx
     }
-    if (containsDriveAlt.nonEmpty) {
-      Some(alternatives(containsDriveAlt.head))
-    } else if (alternatives.nonEmpty) {
-      Some(alternatives(chooseRandomAlternativeIndex(alternatives)))
-    } else {
-      None
-    }
+    containsDriveAlt.headOption
+      .map(idx => alternatives(idx))
+      .orElse(chooseRandomElement(alternatives))
```

Comment:
This comment was lost in a change I think:

chooseRandomElement changes the logic itself. The random index throws explicitly, so now we are hiding this until later. If kept, we should remove that method altogether as it is only used in one other location anyway.

Reply:
I am catching up on all comments and now see your reply is:

> this is the same tradeoff as basically all the changes

While I agree then my latter comment comes into play - we should remove the index method altogether - or make it have the same semantics

Reply:
Lets the the usage of previous method:
```
class ModeChoiceUniformRandom(val beamConfig: BeamConfig) extends ModeChoiceCalculator {

  override def apply(
    alternatives: IndexedSeq[EmbodiedBeamTrip],
    attributesOfIndividual: AttributesOfIndividual,
    destinationActivity: Option[Activity],
    person: Option[Person] = None
  ): Option[EmbodiedBeamTrip] = {
    if (alternatives.nonEmpty) {
      Some(alternatives(chooseRandomAlternativeIndex(alternatives)))
    } else {
      None
    }
  }
```